File: app/src/main/java/com/aquamorph/frcmanager/models/Tab.java
package com.aquamorph.frcmanager.models;


import android.support.v4.app.Fragment;

/**
 * Created by aquam on 2/22/2018.
 */

public class Tab implements Comparable {
	public String name = "";
	public Fragment fragment;

	public Tab(String name, Fragment fragment) {
		this.name = name;
		this.fragment = fragment;
	}


	@Override
	public int compareTo(Object another) {
		return getTabOrder(this.name) - getTabOrder(((Tab) another).name);
	}

	private int getTabOrder(String name) {
		switch (name) {
			case "Team Schedule":
				return 0;
			case "Event Schedule":
				return 1;
			case "Rankings":
				return 2;
			case "Teams":
				return 3;
			case "Alliances":
				return 4;
			case "Awards":
				return 5;
			default:
				return 10;
		}
	}
}

File: app/src/main/kotlin/com/aquamorph/frcmanager/models/tba/Tab.kt
package com.aquamorph.frcmanager.models.tba

import androidx.fragment.app.Fragment

/**
 * Store information about tabs.
 *
 * @author Christian Colglazier
 * @version 3/30/2018
 */

class Tab(name: String, var fragment: Fragment) : Comparable<Tab> {
    var name = ""

    init {
        this.name = name
    }

    override operator fun compareTo(other: Tab): Int {
        return getTabOrder(this.name) - getTabOrder(other.name)
    }

    private fun getTabOrder(name: String): Int {
        return when (name) {
            "Team Schedule" -> 0
            "Event Schedule" -> 1
            "Rankings" -> 2
            "Teams" -> 3
            "Alliances" -> 4
            "Awards" -> 5
            else -> 10
        }
    }
}
